### PR TITLE
Make model somewhat compatible with sqlalchemy-1.4

### PR DESCRIPTION
**Pull request by mvdbeek**
Replaces deprecated joinedload_all, removes trace logging via the proxy
argument to create_engine and re-arranging the mapper classes
to avoid unmapped class exceptions:
```
Traceback (most recent call last):
  File "/Users/mvandenb/.pyenv/versions/3.8.2/lib/python3.8/runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/mvandenb/.pyenv/versions/3.8.2/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/mvandenb/.vscode/extensions/ms-python.python-2020.11.371526539/pythonFiles/lib/python/debugpy/__main__.py", line 45, in <module>
    cli.main()
  File "/Users/mvandenb/.vscode/extensions/ms-python.python-2020.11.371526539/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 430, in main
    run()
  File "/Users/mvandenb/.vscode/extensions/ms-python.python-2020.11.371526539/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 267, in run_file
    runpy.run_path(options.target, run_name=compat.force_str("__main__"))
  File "/Users/mvandenb/.pyenv/versions/3.8.2/lib/python3.8/runpy.py", line 263, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/Users/mvandenb/.pyenv/versions/3.8.2/lib/python3.8/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/Users/mvandenb/.pyenv/versions/3.8.2/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/mvandenb/src/galaxy/scripts/fapi.py", line 14, in <module>
    from galaxy.webapps.galaxy.buildapp import app_factory
  File "/Users/mvandenb/src/galaxy/lib/galaxy/webapps/galaxy/buildapp.py", line 12, in <module>
    import galaxy.app
  File "/Users/mvandenb/src/galaxy/lib/galaxy/app.py", line 10, in <module>
    from galaxy import config, job_metrics, jobs
  File "/Users/mvandenb/src/galaxy/lib/galaxy/config/__init__.py", line 30, in <module>
    from galaxy.model import mapping
  File "/Users/mvandenb/src/galaxy/lib/galaxy/model/mapping.py", line 2344, in <module>
    exists([model.HistoryDatasetAssociation],
  File "<string>", line 2, in exists
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/sql/selectable.py", line 5644, in __init__
    s = Select._create(*args, **kwargs).scalar_subquery()
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/sql/selectable.py", line 4489, in _create
    return cls.create_legacy_select(*args, **kw)
  File "<string>", line 2, in create_legacy_select
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/util/deprecations.py", line 369, in warned
    return fn(*args, **kwargs)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/sql/selectable.py", line 4371, in create_legacy_select
    self._raw_columns = [
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/sql/selectable.py", line 4372, in <listcomp>
    coercions.expect(
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/sql/coercions.py", line 148, in expect
    insp._post_inspect
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 1123, in __get__
    obj.__dict__[self.__name__] = result = self.fget(obj)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/orm/mapper.py", line 2103, in _post_inspect
    configure_mappers()
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/orm/mapper.py", line 3369, in configure_mappers
    mapper._post_configure_properties()
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/orm/mapper.py", line 1896, in _post_configure_properties
    prop.init()
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/orm/interfaces.py", line 226, in init
    self.do_init()
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/orm/relationships.py", line 2134, in do_init
    self._process_dependent_arguments()
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/orm/relationships.py", line 2210, in _process_dependent_arguments
    self.target = self.entity.persist_selectable
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 1056, in __get__
    obj.__dict__[self.__name__] = result = self.fget(obj)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/orm/relationships.py", line 2105, in entity
    return mapperlib.class_mapper(argument, configure=False)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/sqlalchemy/orm/base.py", line 443, in class_mapper
    raise exc.UnmappedClassError(class_)
sqlalchemy.orm.exc.UnmappedClassError: Class
'galaxy.model.HistoryDatasetAssociationTagAssociation' is not mapped
```

### CODE REVIEW

**Comment by mvdbeek** [comment]
Hmm, the selenium error looks real.

File: lib/galaxy/model/orm/engine_factory.py [hunk 1]
```diff
@@ -25,16 +27,6 @@ def log_request_query_counts(req_id):
 
 
 def build_engine(url, engine_options, database_query_profiling_proxy=False, trace_logger=None, slow_query_log_threshold=0, thread_local_log=None, log_query_counts=False):
-    # Should we use the logging proxy?
-    if database_query_profiling_proxy:
```
Comment (jmchilton) [inline lib/galaxy/model/orm/engine_factory.py:29]: Shouldn't we drop the option or implement a replacement?

Comment (mvdbeek) [reply lib/galaxy/model/orm/engine_factory.py:29]: I think a replacement would look like the slow query logger below, the `ConnectionProxy` that logging_connection_proxy uses has been removed (https://docs.sqlalchemy.org/en/14/changelog/changelog_14.html?highlight=connectionproxy#change-6daf2f59ac2438ef9c8213e9ebeac157)

Comment (mvdbeek) [reply lib/galaxy/model/orm/engine_factory.py:29]: In fact setting `slow_query_log_threshold`  to 0 should be almost equivalent.

Comment (mvdbeek) [reply lib/galaxy/model/orm/engine_factory.py:29]: I've added a replacement in https://github.com/galaxyproject/galaxy/pull/10856/commits/23852b22d771e928e169c46858c9b79bab0538c7

File: lib/galaxy/model/orm/engine_factory.py [hunk 2]
```diff
@@ -3,7 +3,9 @@
 import time
 
 from sqlalchemy import create_engine, event
```
Comment (nsoranzo) [inline lib/galaxy/model/orm/engine_factory.py]:
```suggestion
from sqlalchemy import (
    create_engine,
    event,
)
```

File: lib/galaxy/model/orm/engine_factory.py [hunk 3]
```diff
@@ -3,7 +3,9 @@
...
-from sqlalchemy.engine import Engine
+from sqlalchemy.engine import (
+    Engine,
+)
```
Comment (nsoranzo) [inline lib/galaxy/model/orm/engine_factory.py]:
```suggestion
from sqlalchemy.engine import Engine
```

**Review by jmchilton — APPROVED** [review]

**Comment by nsoranzo** [comment]
@mvdbeek The [`database_query_profiling_proxy` option help](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/config_schema.yml#L107) is still referring to `'galaxy.model.orm.logging_connection_proxy'` , can you update it please?

**Comment by mvdbeek** [comment]
Yes, will do

**Pull request merged**